pkgs/by-name/wh/where-is-my-sddm-theme/package.nix

```diff
@@ -33,13 +33,13 @@ lib.checkListOfEnum "where-is-my-sddm-theme: variant" validVariants variants
 
 stdenvNoCC.mkDerivation rec {
   pname = "where-is-my-sddm-theme";
-  version = "1.9.1";
+  version = "1.9.2";
 
   src = fetchFromGitHub {
     owner = "stepanzubkov";
-    repo = pname;
-    rev = "v${version}";
-    hash = "sha256-o9SpzSmHygHix3BUaMQRwLvgy2BdDsBXmiLDU+9u/6Q=";
+    repo = "where-is-my-sddm-theme";
```

Comment:
What is the point of hardcoding `repo`? If you want to avoid `rec`, then you forgot remove it. In any case, simple derivations like this are not a place where "antipatterns" can lead to serious problems. Also, there is no code style convention in nixpkgs, so there is no reason to change the code style.

All you need for update is change version and hashes like @r-ryantm does

Reply:
The style convention is to use `nixfmt-rfc-style` now (see RFC140).

Re-using `pname` is a bad idea, it creates a binding that goes too far. In case of `pname` attribute change, the derivation is broken. This is not a good practice to reuse it.

Reply:
In the past, I got told by maintainers to switch `pname` to hardcoding the name of the repo. This is why I change it in the packages I edit.

Reply:
Using a plain ref works but is ambiguous and can conflict with a branch named the same.

Reply:
> The style convention is to use nixfmt-rfc-style now (see RFC 0166), but I personally enforce it only on new files to avoid dealing with a huge diff.

This was already formatted using `nixfmt-rfc-style`. When I said code style, I meant code patterns, not formatting style.

> What did Gaetan is fine, I don't think we should spend more time discussing these details.

I'm not saying it's bad, it just didn't make sense. And if he wants to fix `antipatterns', he should go all the way and remove `rec` as well. I'm not against all these changes.

Reply:
> I'm not saying it's bad, it just didn't make sense. And if he wants to fix `antipatterns', he should go all the way and remove `rec` as well. I'm not against all these changes.

We still need `rec` for the version if I am not mistaking.

Reply:
Not necessarily, we could simply use a `let...in` clause.

Reply:
> still need

just pass a function to `mkDerivation`

https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes

Reply:
> > still need
> 
> just pass a function to `mkDerivation`

Oh indeed, the `finalAttrs` pattern!

Reply:
Done. + reformatted with `nixfmt-rfc-style`, because it was already formatted this way prior to this PR.